docs/operating.adoc

```diff
@@ -27,7 +27,72 @@ KROXYLICIOUS_ROOT_LOG_LEVEL="DEBUG"
 
 to configure the root log level (note this will be very verbose at DEBUG/TRACE)
 
-== Configuring TLS
+== Authentication
+
+There are two authentications to consider :
+
+* Downstream: Client to Kroxy
+* Upstream: Kroxy to Kafka
+
+=== Downstream authentication
+
+==== PLAIN
+
+#### Kafka client configuration
+
+[source]
+----
+# Configure SASL_SSL if TLS/SSL encryption is enabled, otherwise configure SASL_PLAINTEXT
+security.protocol=SASL_SSL
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
+    username="USERNAME" \
+    password="PASSWORD";
+----
+
+#### Virtual cluster configuration
+
+WARNING: At this time PLAIN is only supported as delegated authentication (managed by upstream)
+
+==== OAUTHBEARER
+
+#### Kafka client configuration
+
+[source]
+----
+# Configure SASL_SSL if TLS/SSL encryption is enabled, otherwise configure SASL_PLAINTEXT
+security.protocol=SASL_SSL
+sasl.mechanism=OAUTHBEARER
+sasl.login.callback.handler.class=org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler
+sasl.oauthbearer.token.endpoint.url=JWKS_ENDPOINT
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  clientId="CLIENT_ID" \
+  clientSecret="CLIENT_SECRET"
+  scope="<scope>"; #<1>
+----
+<1> optional scope
+
+#### Virtual cluster configuration
+
+Setting OAuth on virtual cluster is optional and will indicate to Kroxy to handle the SASL connection.
+The SASL connection is fully managed by Kroxy, upstream must not require a SASL connection because handshake
+and authenticate requests are not forwarded to upstream.
+Currently, Kroxylicious doesn't manage RBAC, so there is no role validation.
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    oauth:                          #<1>
+      jwksEndpointUrl: <url>        #<2>
```

Comment:
Bikeshedding on the configuration schema for a moment... we need to think how the configuration can evolve to suit future needs. Kafka itself supports a number of SASL mechanisms. The proxy adds the possibility of handling authentication on the proxy or delegating/propagating the SASL requests to the broker. Ignoring for a moment the possibility multiple listeners on the broker, I feel that this is closer to what we should aim to express:

```suggestion
virtualClusters:
  demo:
    sasl:
      plain: 
        policy: disable  # 1
      scramSha256: 
        policy: propagate # 2
      scramSha512: 
        policy: propagate
      oauthBearer: # 3
        policy: authenticate
        configuration:
          jwksEndpointUrl: <url>
```

1. A mechanism can be in `disable` mode. Setting disabled means the mechanism won't allowed. If a client sends this mechanism in a `SaslHandshake` request we'll return `UNSUPPORTED_SASL_MECHANISM`.
2. A mechanism can be in `propagate` mode. Setting `propagate` means the mechanism will be allowed and the handler will forward `SaslAuthenticate` requests to the broker.
3. A mechanism can be in `authenticate` mode. This means the mechanism will be allowed but the handler will handle the `SaslAuthenticate` requests. In this mode we also have to supply the configuration necessary for the proxy to handle the authentication itself. 

For backwards compatibility if a mechanism is omitted then it's considered to be in `propagate` mode. If `sasl` itself is omitted then we propagate everything to the broker.